pkg/reconciler/apis/apibindingdeletion/apibinding_deletion_controller.go

```diff
@@ -111,7 +112,7 @@ type Controller struct {
 }
 
 func (c *Controller) enqueue(obj interface{}) {
-	key, err := cache.MetaNamespaceKeyFunc(obj)
+	key, err := kcpcache.MetaClusterNamespaceKeyFunc(obj)
```

Comment:
when does this have to be the deletion handling variant?

Reply:
In order to be more certain about the automated refactor code, I did not make any opinionated choice there - a simple change from non-cluster-aware to cluster-aware is the refactor. If we think it's a good idea, though, at least in our code I can move stuff to be deletion handling aware in a follow-up.

Reply:
DeletionHandling should be used on objects received directly as params to AddFunc/UpdateFunc/DeleteFunc. You don't _have_ to, but it does mean that if the reflector missed the delete event and it relists and the object is gone, you'll get a DeletedFinalStateUnknown, in which case you'll need to use the deletion handling variant to determine the key. And if you want the object in its last known state, you'll need to properly handle the tombstone.